### PR TITLE
Add project view details page

### LINKED ISSUE #451: [DSG]: Support Project View
### Feature description

We need a page that can support applying multiple actions under your project view ([mocks](https://marvelapp.com/prototype/afa716d/screen/88098881)).

- Create a view that has a breadcrumb to return to the project list page
- Add stacks multiple sections
    - "Data science workspaces"
    - "Storage"
    - "Data connections"
    - Ignore "Model serving" for now
- Make sure each section has an empty state -- ignore the secondary action button for now
- Add a "jump to section" nav control on the left
- Ignore the tab at the top for now

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Resolves: #451 

## Description

Missing breadcrumb nav for now because @dlabaj will implement that in #560 
Created a page for project details with layout and only empty states in it.

Screen Shot 2022-09-20 at 9 51 00 AM

## How Has This Been Tested?

Go to `${dashboardURL}/projects/${projectName}` and you will find this page. There is currently no check for the URL so anything you input for the `projectName` in the URL will go to this page. We can add a hook to search for this project based on the URL and only show the page if the project exists.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const ProjectDetailsSidebarWrapper: React.FC = ({ children }) => {
+  return (
+    <Sidebar hasGutter>
+      <SidebarPanel variant="sticky">
+        <JumpLinks isVertical label="Jump to section" scrollableSelector="#project-details-list">
+          <JumpLinksItem href="#data-science-workspaces">Data science workspaces</JumpLinksItem>
+          <JumpLinksItem href="#storage">Storage</JumpLinksItem>
+          <JumpLinksItem href="#data-connections">Data connections</JumpLinksItem>
+          <JumpLinksItem href="#model-serving">Model serving</JumpLinksItem>
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx:18]: Can we enum these values so it's less magic string?

File: frontend/src/pages/projects/screens/detail/WorkspacesList.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          </FlexItem>
+        </Flex>
+      </StackItem>
+      <StackItem>
+        <EmptyDetailsList
+          title="No data science workspaces"
+          description="To get started, create a data science workspace."
+        />
+      </StackItem>
+    </Stack>
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/WorkspacesList.tsx:26]: No way to DRY this? You have a lot of repeat boilerplate code.

What about something like:
```
<DetailsSection
  id={...the enum you could create...}
  title="..."
  actions={[
    <Button key="create" variant="secondary">...</Button>,
  ]}
>
  <EmptyDetailsList title="..." description="..." />
</DetailsSection>
```

When we get into the data states, we can just do a ternary around the `EmptyDetailsList` component and render the data (or maybe we have a wrapper there too, and this is just a prop to THAT component). Anyway we cut this, we should look to be more detailed with our intentions.

File: frontend/src/pages/projects/screens/detail/ProjectDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { Divider, PageSection, Stack, StackItem } from '@patternfly/react-core';
+import * as React from 'react';
+import { useParams } from 'react-router';
+import ApplicationsPage from '../../../ApplicationsPage';
+import EmptyProjects from '../../EmptyProjects';
+import DataConnectionsList from './DataConnectionsList';
+import ModelServingList from './ModelServingList';
+import ProjectDetailsSidebarWrapper from './ProjectDetailsSidebar';
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/ProjectDetails.tsx:8]:
```suggestion
import ProjectDetailsSidebar from './ProjectDetailsSidebar';
```
No need to name it something other than what it is

File: frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import {
+  JumpLinks,
+  JumpLinksItem,
+  Sidebar,
+  SidebarContent,
+  SidebarPanel,
+} from '@patternfly/react-core';
+
+const ProjectDetailsSidebarWrapper: React.FC = ({ children }) => {
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx:10]:
```suggestion
const ProjectDetailsSidebar: React.FC = ({ children }) => {
```
Components should share the name of the file

File: frontend/src/pages/projects/screens/detail/ProjectDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+              <StorageList />
+            </StackItem>
+            <Divider />
+            <StackItem>
+              <DataConnectionsList />
+            </StackItem>
+            <Divider />
+            <StackItem>
+              <ModelServingList />
+            </StackItem>
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/ProjectDetails.tsx:40]: This is not for now -- this is for the next release.

File: frontend/src/pages/projects/screens/detail/ProjectDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      description={null}
+      loaded
+      empty={false}
+      emptyStatePage={<EmptyProjects />}
+    >
+      <PageSection id="project-details-list" hasOverflowScroll variant="light">
+        <ProjectDetailsSidebarWrapper>
+          <Stack hasGutter>
+            <StackItem>
+              <WorkspacesList />
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/ProjectDetails.tsx:27]: Almost want to just map over them... boilerplate between each one is kinda error prone.

```
const sections = [
  // component could even be data based on my previous comment about boilerplate code 
  { id: '...enum value...', component: <WorkspacesList /> },
  { id: '...enum value...', component: <StorageList /> },
  ...
];

<Stack hasGutter>
  {sections.map(({ id, component }) => (
    <React.Fragement key={id}>
      <StackItem>
        {component}
      </StackItem>
      <Divider />
    </React.Fragment>
  )}
</Stack>
```

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Looking good -- I think we can reduce some of the boilerplate though. Let me know what you think.

File: frontend/src/pages/projects/screens/detail/types.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+export enum ProjectSectionID {
+  WORKSPACE = 'data-science-workspaces',
+  STORAGE = 'storage',
+  DATA_CONNECTIONS = 'data-connections',
+  MODEL_SERVING = 'model-serving',
+}
+
+export enum ProjectSectionTitle {
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/types.ts:8]: Why not make this a map to labels -- that way TypeScript can make sure we are looking at things correctly?

image

```
enum Test {
  ONE = 'alpha',
  TWO = 'beta',
}

type TestLabels = { [key in Test]: string };

const TestLabels: TestLabels = {
  [Test.ONE]: 'Alpha Label',
}

const alphaLabel = TestLabels[Test.ONE];
```

File: frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  SidebarContent,
+  SidebarPanel,
+} from '@patternfly/react-core';
+import { ProjectSectionID, ProjectSectionTitle } from './types';
+
+const ProjectDetailsSidebar: React.FC = ({ children }) => {
+  const navItems = [
+    { id: ProjectSectionID.WORKSPACE, title: ProjectSectionTitle.WORKSPACE },
+    { id: ProjectSectionID.STORAGE, title: ProjectSectionTitle.STORAGE },
+    { id: ProjectSectionID.DATA_CONNECTIONS, title: ProjectSectionTitle.DATA_CONNECTIONS },
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx:15]: With my other comments, this has to be less something you manually craft, and more something you use as a lookup map.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+  return (
+    <Sidebar hasGutter>
+      <SidebarPanel variant="sticky">
+        <JumpLinks isVertical label="Jump to section" scrollableSelector="#project-details-list">
+          {jumpLinkItems.map(({ id, title }) => (
+            <JumpLinksItem key={`link-${id}`} href={`#${id}`}>
+              {title}
+            </JumpLinksItem>
+          ))}
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx:30]: Please avoid creating extra items you can extrapolate from anyways. Data is best done when there is only a single source of truth. Untyped Arrays & Object creation are also very bad (`jumpLinkItems`), avoid those as much as you can. Maybe we can do something with TS to make it yell at untyped items.

Suggested change:
```
{Object.values(ProjectSectionID).map((section) => (
  <JumpLinksItem key={section} href={`#${section}`}>
    {ProjectSectionTitles[section]}
  </JumpLinksItem>
)}
```

File: frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx (hunk 8 above)
Comment (andrewballantyne) [reply frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx]: I gave more details here: https://github.com/opendatahub-io/odh-dashboard/pull/578#discussion_r975795618

File: frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx (hunk 9 above)
Comment (DaoDaoNoCode) [reply frontend/src/pages/projects/screens/detail/ProjectDetailsSidebar.tsx:30]: I did try this method but I used it in a wrong way at that time and rendered nothing 😞 now it works, thanks

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**